pkg/admission/certwriter/doc.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package certwriter provide method to handle the CA and certificate for webhooks.
+
+Create a default CertWriterProvider
+
+	writerProvider, err := NewProvider(Options{client: client}))
+	if err != nil {
+		// handler error
+	}
+
+Create a SecretCertWriterProvider
```

Comment:
Why this level of indirection - why not simply have this take the webhookConfiguration?

Reply:
Or better yet, have `EnsureCert` take the configuration - e.g. `EnsureCert(webhookConfiguration)`